### DISCUSSION TITLE
Is there any way to retrieve factory instance in scope of the test

### DISCUSSION THREAD

**Discussion by vopolonc** [worksome/request-factories#40] [Q&A]
Is there a way to retrieve the factory instance (or faked data) in the test scope? It would be handy to use it for assertions and to compare responses:

```PHP
it('can update addons', function () {
    $this->patch($this->endpointOrder($this->order->id))
        ->assertOk()
        ->assertJsonFragment(['addons' => $this->fakeFactory->data['addons']);
})
    ->fakeRequest(UpdateRequestFactory::class)
    ->addons();
```

Maybe I'm missing something, let me know

**Comment by dcaswel** [comment]
I usually use `UpdateRequestFactory::create()` and assign that to a variable that I pass into `patch()` instead of using `fakeRequest()`. Then you have it available for assertions. It would be nice if there was some way to access it while using `fake()` or `fakeRequest()` though.

**Reply by vopolonc** [reply]
For now, I'm doing the exact same thing and trying to figure out a more convenient way

**Comment by Livijn** [comment]
The preferred way for me would be `->fake()` would return the data used from the factory. I'll continue with the `create`-method instead and passing the data to the request.
